File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int no_z_prod = 1;
        int c = 0;
        for(auto i:nums)
        {
            if(i==0)
                ++c;
            else
            {
                no_z_prod *=i;
            }
        }
        vector<int> res;
        if(c==0)
        {
            for(auto i:nums)
                res.push_back(no_z_prod/i);
        }
        else if(c==1)
        {
            for(auto i:nums){
                if(i==0)
                    res.push_back(no_z_prod);
                else
                    res.push_back(0);
            }
        }
        else{
            for(auto i:nums)
                res.push_back(0);
        }
        return res;
    }
};

File: Data Structures & Algorithms/majority-element-ii/submission-0.cs

public class Solution {
    public List<int> MajorityElement(int[] nums) {
        Dictionary<int, int> freq_map = new Dictionary<int, int>();
        int n = nums.Length;
        List<int> res = new List<int>();
        foreach (int i in nums)
        {
            if (!freq_map.ContainsKey(i)) {
                freq_map[i] = 0;
            }
            freq_map[i]++;
            if (freq_map[i] > n / 3 && !res.Contains(i))
            {
                res.Add(i);
            }
        }
        return res; 
    }
};